android/src/main/java/com/learnium/RNDeviceInfo/RNDeviceModule.java

```diff
@@ -454,11 +455,13 @@ public boolean isBatteryChargingSync(){
   public void isBatteryCharging(Promise p) { p.resolve(isBatteryChargingSync()); }
 
   @ReactMethod(isBlockingSynchronousMethod = true)
-  public int getUsedMemorySync() {
-    Runtime rt = Runtime.getRuntime();
-    long usedMemory = rt.totalMemory() - rt.freeMemory();
-    return (int)usedMemory;
+  public double getUsedMemorySync() {
+    Debug.MemoryInfo memInfo = new Debug.MemoryInfo();
```

Comment:
Interesting! First, thanks for looking into this, I was unaware there was a discrepancy in here, no one has mentioned it before!

Have you considered the warning here? https://developer.android.com/reference/android/os/Debug#getMemoryInfo(android.os.Debug.MemoryInfo)

> Note: this method directly retrieves memory information for the given process from low-level data available to it. It may not be able to retrieve information about some protected allocations, such as graphics. If you want to be sure you can see all information about allocations by the process, use ActivityManager.getProcessMemoryInfo(int[]) instead.

It appears this is the recommended way to make sure you really get accurate info: https://developer.android.com/reference/android/app/ActivityManager#getProcessMemoryInfo(int[])

Appears you can easily get the PID like so https://developer.android.com/reference/android/os/Process#myPid() ?

Reply:
@mikehardy I'm not 100% if that API is better or not. It notes some restrictions around rate limits and how it's only intended for debugging 

> Note: this method is only intended for debugging or building a user-facing process management UI.

I feel like the API I'm using should be sufficient.

I'm very open to feedback and it is your repo so say the magic words and I will swap it out. Thanks for the feedback!

Reply:
I see what you mean... And yet I think graphics is a frequent consumer of lots of memory. Also I don't think people will call this much? I usually lean towards exposing all the info possible so I think the more thorough API is better, but maybe a Readme update with a link to the android API and the warning text? So developers know there may be a real cost.

Reply:
@tbezman This is not meant to pressure: I'm just sweeping my mailbox of all notifications and I'd saved this one. Just to be clear this is waiting on swapping out the API for the one that contains graphic memory as well, along with maybe a readme note containing the admonition from API docs and a pointer to them so devs can make their tradeoff decisions. Then I'll merge it and it'll auto-release and be amazing :sweat_smile:

Reply:
Hey @mikehardy. Just now getting back to this. I'll have a commit up soon!